spi/src/main/java/org/opensearch/jobscheduler/spi/schedule/CronSchedule.java

```diff
@@ -172,17 +187,19 @@ public boolean equals(Object o) {
         if (o == null || getClass() != o.getClass()) return false;
         CronSchedule cronSchedule = (CronSchedule) o;
         return timezone.equals(cronSchedule.timezone) &&
-                expression.equals(cronSchedule.expression);
+                expression.equals(cronSchedule.expression) &&
+                scheduleDelay == cronSchedule.scheduleDelay;
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(timezone, expression);
+        return Objects.hash(timezone, expression, scheduleDelay);
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeZoneId(timezone);
         out.writeString(expression);
+        out.writeOptionalLong(scheduleDelay);
```

Comment:
does this need to be optional? since the class is using the primitive type for the `scheduleDelay` this field should always be present

Reply:
This definitely didn't need to be optional with the primitive scheduleDelay, but I have since changed scheduleDelay to be a Long object as part of making the job scheduler backwards compatible. Now, if scheduleDelay is null then it isn't included in the streams or jsons, so it shouldn't blow up the other plugins which use the job scheduler without a delay. The downside is that in a plugin using delay, you would always need to use the constructor with the delay, even if you are just setting it to 0, as otherwise the streams and jsons wouldn't include the delay value.

Reply:
> so it shouldn't blow up the other plugins which use the job scheduler without a delay
What do you mean by this - how will this affect other plugins? the logic of constructing and deconstructing the objects is limited to just these classes not sure I am seeing what is the concern with using `long` and including it in bytestream always?